### PR TITLE
Set better date picker defaults

### PR DESCRIPTION
**Pull request by pauldambra**
## Changes

This 

* sets "before" as the default date operator
* removes operators that don't make sense with dates
* adds quick choices to the date picker

While it uses the insight date filter relative dates as a source. Unlike the insight date filters, it does not send the relative date string to the API. This was to avoid changing PropertyFilters which don't have an API value and display value

## How did you test this code?

running it locally and seeing that the default operator is before and quick choices work as expected

### CODE REVIEW

File: frontend/src/lib/components/PropertyFilters/components/OperatorValueSelect.tsx [hunk 1]
```diff
@@ -45,11 +45,21 @@ export function OperatorValueSelect({
...
 }: OperatorValueSelectProps): JSX.Element {
-    const [currentOperator, setCurrentOperator] = useState(operator)
     const { propertyDefinitions } = useValues(propertyDefinitionsModel)
 
     const propertyDefinition = propertyDefinitions.find((pd) => pd.name === propkey)
 
+    // DateTime properties should not default to Exact
+    const startingOperator =
+        allowQueryingEventsByDateTime &&
+        propertyDefinition?.property_type == 'DateTime' &&
```
Comment (mariusandra) [inline frontend/src/lib/components/PropertyFilters/components/OperatorValueSelect.tsx:55]: Hmm, I would have thought `tsc` would flag this:

```suggestion
        propertyDefinition?.property_type == PropertyType.DateTime &&
```

File: frontend/src/lib/components/PropertyFilters/components/OperatorValueSelect.tsx [hunk 2]
```diff
@@ -45,11 +45,21 @@ export function OperatorValueSelect({
...
+        (!operator || operator == PropertyOperator.Exact)
+            ? PropertyOperator.IsDateBefore
+            : operator
+            ? operator
+            : PropertyOperator.Exact
```
Comment (mariusandra) [inline frontend/src/lib/components/PropertyFilters/components/OperatorValueSelect.tsx:60]: This is trippy and hard to follow 🤪. Could `: (operator || PropertyOperator.Exact)` make anything cleaner instead of another `?:`?

**Review by mariusandra — CHANGES_REQUESTED** [review]
The quick choices are nice! Few other points:

- It appears I can't edit a date once it has been set

- If you just store days as properties `2020-06-26`, you might want to use the "equal" operator. However... I don't know with all the timezone conversions, if this would actually match things precisely now. If you try, you'll have the Greenwich "bug" when testing :D... Don't ask how often it's been useful to know that `2020-06-26 23:00:00 !== 2020-06-27 00:00:00` when debugging these issues. 
- Is `before` the best default? I'd intuitively reach for `after` or as the first thing... 🤔 . Allowing ranges would be best though.

File: frontend/src/lib/components/PropertyFilters/components/OperatorValueSelect.tsx (hunk 2 above)
Comment (pauldambra) [reply frontend/src/lib/components/PropertyFilters/components/OperatorValueSelect.tsx:60]: It wasn't _too_ bad before Prettier reformated it... but, yep

**Comment by pauldambra** [comment]
> Is before the best default? I'd intuitively reach for after or as the first thing... 🤔 . Allowing ranges would be best though.

Since the view starts with most recent event, starting with an after filter only lets you remove the tail (i.e. the bottom of the screen). Before lets you remove the head.

But 🤷‍♀️

----

Was more to remove exact as the default.

I wanted to avoid decisions (in the short term at least) like is the time 14:55:23 equal to 14:55:24, or is 14:55:23.23466?

Can add range as a follow up? Was trying to get to a place where we can get this in front of beta users...

**Comment by mariusandra** [comment]
Looks good. OK to do what's above as a followup...

The point remains though, that when I have a property with the format "2020-05-20", I'd like to be able to filter by this exact date.

**Comment by pauldambra** [comment]
> The point remains though, that when I have a property with the format "2020-05-20", I'd like to be able to filter by this exact date.

🤔
